### DISCUSSION TITLE
Quarkus project as a dependency for other microservices

### DISCUSSION THREAD

**Discussion by fenix-hub** [quarkusio/quarkus#25532] [Q&A]
Hi all,
Is it possible to compile a quarkus project as a maven dependency for other quarkus projects?
Imagine project `A` and project `B` are two independent Quarkus microservices (their own endpoints, schemas and db).
Service `B` wants to directly communicate over HTTP with service `A` using their endpoints and DTOs classes.
Is it possible to use service `A` as a dependency for service `B`, without actually starting service `A` *inside* service B, but only importing service A's classes from its package, while service `A` is running on a different port/server?
Thanks

**Comment by fenix-hub** [comment]
Any update?

**Reply by plevart** [reply]
Hi, with build tool like Maven, this is very easy to do and I practice this often. What I usually do is have those closely related "projects" being just modules in one multi-module Maven project, but that is not the only way - you could have two or more independent Maven projects. Say modules A and B are microservices that include, in their pom.xml, the quarkus-maven-plugin invocation which builds and packs the final products (jars, docker images, native executables). Classes and resources that are common to both A and B you then extract into a normal "library" module C (without invocation of quarkus-maven-plugin in its pom.xml) and add a dependency on that module into A's and B's pom.xml files. You use those classes in C from A and B as though C was a library.
I recommend that A, B and C are located in the same Maven project so that you can have a single GiT repository and single versioning of all interdependencies, but you could also do it differently, say C being in the same Mavem project together with A, while B in a separate Maven project. Or even C in a separate Maven project. Whichever works for you best. If you have interdependencies among Maven projects you then have to resort to Maven versioning of artefacts and, depending on build procedures, deployment of such dependencies to some Maven repository so that they can be downloaded in another Maven project. It's up to you.
Hope this helps.
